File: app/src/main/java/ru/geekbrains/notes/data/CardDataTranslate.java
package ru.geekbrains.notes.data;

import com.google.firebase.Timestamp;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class CardDataTranslate {
    public static class Fields {
        public final static String TITLE = "title";
        public final static String DESCRIPTION = "description";
        public final static String DATE = "date";
    }

    public static CardData documentToCardData(String id, Map<String, Object> doc) {
        CardData result = new CardData(
                (String)doc.get(Fields.TITLE),
                (String)doc.get(Fields.DESCRIPTION),
                ((Timestamp) Objects.requireNonNull(doc.get(Fields.DATE))).toDate());

        result.setId(id);

        return result;
    }

    public static Map<String, Object> cardDataToDocument(CardData cardData) {
        Map<String, Object> result = new HashMap<>();
        result.put(Fields.TITLE, cardData.getTitle());
        result.put(Fields.DESCRIPTION, cardData.getText());
        result.put(Fields.DATE, cardData.getDate());
        return result;
    }
}

File: app/src/main/java/ru/geekbrains/notes/data/CardDataTranslate.kt
package ru.geekbrains.notes.data

import com.google.firebase.Timestamp
import java.util.*

object CardDataTranslate {
    fun documentToCardData(id: String, doc: Map<String, Any?>): CardData {
        return CardData(
            id,
            doc[Fields.TITLE] as String,
            doc[Fields.TEXT] as String,
            (Objects.requireNonNull(doc[Fields.DATE]) as Timestamp).toDate()
        )
    }

    fun cardDataToDocument(cardData: CardData?): Map<String, Any> {
        val result: MutableMap<String, Any> = HashMap()
        if (cardData != null) {
            result[Fields.TITLE] = cardData.title
            result[Fields.TEXT] = cardData.text
            result[Fields.DATE] = cardData.date
        }
        return result
    }

    object Fields {
        const val TITLE = "title"
        const val TEXT = "description"
        const val DATE = "date"
    }
}
